src/Uno.UI/UI/Xaml/Internal/PointerCapture.cs

```diff
@@ -0,0 +1,346 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.Devices.Input;
+using Windows.UI.Xaml.Input;
+
+using Uno.Extensions;
+using Uno.Foundation.Logging;
+using Uno.UI.Extensions;
+using System.Runtime.CompilerServices;
+using Windows.UI.Xaml;
+
+namespace Uno.UI.Xaml.Core;
+
+/*
+ *	This file is intrinsically linked to the UIElement and uses some of its internal APIs:
+ *		internal List<Pointer> PointerCapturesBackingField;
+ *		internal partial void CapturePointerNative(Pointer pointer);
+ *		internal partial void ReleasePointerNative(Pointer pointer);
+ */
+
+/// <summary>
+/// This is an helper class that use to manage the captures for a given pointer.
+/// </summary>
+internal class PointerCapture
+{
+	private static readonly IDictionary<PointerIdentifier, PointerCapture> _actives = new Dictionary<PointerIdentifier, PointerCapture>(EqualityComparer<PointerIdentifier>.Default);
+
+	/// <summary>
+	/// Current currently active pointer capture for the given pointer, or creates a new one.
+	/// </summary>
+	/// <param name="pointer">The pointer to capture</param>
+	public static PointerCapture GetOrCreate(Pointer pointer)
+		=> _actives.TryGetValue(pointer.UniqueId, out var capture)
+			? capture
+			: new PointerCapture(pointer); // The capture will be added to the _actives only when a target is added to it.
+
+	internal static bool TryGet(PointerIdentifier pointer, out PointerCapture capture)
+		=> _actives.TryGetValue(pointer, out capture);
+
+	public static bool TryGet(Pointer pointer, out PointerCapture capture)
+		=> _actives.TryGetValue(pointer.UniqueId, out capture);
+
+	public static bool Any(out List<PointerCapture> cloneOfAllCaptures)
+	{
+		if (_actives.Any())
+		{
+			cloneOfAllCaptures = _actives.Values.ToList();
```

Comment:
Do we really need `List<T>`? Is `Values` (`ICollection<T>`) not sufficient?

Reply:
Theoretically yes, but there is a usage in the code where we are using the indexer on `cloneOfAllCaptures`. As it's internal code, I think it's better to give "full" access on it to avoid cloning the list again and again.